includes/payment-methods/class-upe-split-payment-gateway.php

```diff
@@ -548,6 +550,33 @@ public function get_stripe_id() {
 		return $this->stripe_id;
 	}
 
+	/**
+	 * Handles the shipping requirement for Afterpay payments.
+	 *
+	 * This method extracts the shipping and billing data from the order and sets the appropriate
+	 * shipping data for the Afterpay payment request. If neither shipping nor billing data is valid
+	 * for shipping, an exception is thrown.
+	 *
+	 * @param WC_Order                     $order    The order object containing shipping and billing information.
+	 * @param Create_And_Confirm_Intention $request The Afterpay payment request object to set shipping data on.
+	 *
+	 * @throws Invalid_Address_Exception If neither shipping nor billing address is valid for Afterpay payments.
+	 * @return void
+	 */
+	protected function handle_afterpay_shipping_requirement( WC_Order $order, Create_And_Confirm_Intention $request ): void {
+		$shipping_data = $this->get_shipping_data_from_order( $order );
+		if ( WC_Payments_Utils::is_data_valid_against_keys( $shipping_data['address'], Afterpay_Payment_Method::SHIPPING_ADDRESS_REQUIRED_FIELDS ) ) {
```

Comment:
There are better approaches than yearly generalization here as we solve extremely narrow task.
We can use something less sophisticated but better contextualized, like:

```php
$check_if_usable = function( array $address ): bool {
	return $address['country'] && $address['state'] && $address['city'] && $address['postal_code'] && $address['line1'];
}

if ( $check_if_usable( $shipping_data['address'] ) ) {
    // ...
}

if ( $check_if_usable( $billing_data['address'] ) ) {
    // ...
}
```

Reply:
Indeed, that was a bit too much. Using suggested approach in this commit https://github.com/Automattic/woocommerce-payments/pull/7541/commits/16c534233b8a75a7c727bd6ec870c87f91da4ea4 Thanks for the good pointers (and patience)